src/v2i-hub/CARMAStreetsPlugin/src/J2735ToSRMJsonConverter.cpp

```diff
@@ -93,20 +97,14 @@ namespace CARMAStreetsPlugin
             }            
             request["expectedTimeOfArrival"] = expectedTimeOfArrival;
 
-            /**
-             * The requestor identifies itself using its current speed, heading and location.
-             */
-            std::stringstream ss;
-            ss << srm_ptr->requestor.id.choice.entityID.buf;
-            auto id_len = srm_ptr->requestor.id.choice.entityID.size;
-            unsigned long id_num = 0;
-            for(auto i = 0; i < id_len; i++)
-            {			
-                id_num = (id_num << 8) | srm_ptr->requestor.id.choice.entityID.buf[i];
+            if(srm_ptr->requestor.id.choice.entityID.size == sizeof(int32_t))
+            {
+                request["msgCount"] = srm_ptr->requestor.id.choice.entityID.size;
```

Comment:
Is it intended here to overwrite msgCount with sizeof(int32_t) or "4" everytime this condition is satisfied? 
Shouldn't the incoming message be preserved as is?